activerecord/test/cases/locking_test.rb

```diff
@@ -738,6 +738,53 @@ def test_with_lock_rolls_back_transaction
       assert_equal old, person.reload.first_name
     end
 
+    def test_with_lock_configures_transaction
+      person = Person.find 1
+      Person.transaction do
+        outer_transaction = Person.connection.transaction_manager.current_transaction
+        assert_equal true, outer_transaction.joinable?
+        person.with_lock(requires_new: true, joinable: false) do
+          current_transaction = Person.connection.transaction_manager.current_transaction
+          assert_not_equal outer_transaction, current_transaction
+          assert_equal false, current_transaction.joinable?
+        end
+      end
+    end
+
+    if current_adapter?(:PostgreSQLAdapter)
```

Comment:
This is a pattern I saw in other tests which feels appropriate here. In order to test transaction behavior I need to do these tests on an engine that supports transactions, but it doesn't feel important to test all database engines' specific behavior at this level, so I chose postgresql.